File: DataStructures/DijkstraSearch/Node.cpp
#include "Node.h"
Node::Node(char val) {
	this->value = val;
}
Node::~Node() {
	for (int i = 0; i < this->neighbors.size();i++) {
		if (this->neighbors[i] != nullptr) {
			delete this->neighbors[i];
		}
		this->neighbors[i]->to = nullptr;
		 this->neighbors[i] = nullptr;
	}
}
void Node::AddRoute(Node* node,float cost) {
	Edge* neighbor = new Edge(node,cost);
	this->neighbors.push_back(neighbor);
}

File: src/graph/depthFirst/node.ts
import {INode} from "../INode";
import {Edge} from "./edge";
export class Node implements INode<string>{
    neighbors: Array<Edge>;
    value: string;

    constructor(val:string) {
        this.neighbors = [];
        this.value = val;
    }

    addRoute(node: Node): void {
        let neighbor = new Edge(node);
        this.neighbors.push(neighbor);
    }

}